src/lua/api-gateway/cache/store/redisCache.lua

```diff
@@ -53,9 +70,60 @@ local function getRedisUpstream(upstream_name)
     return nil, nil
 end
 
-local cache_store_cls = require "api-gateway.cache.store"
+---
+--- @params upstream
+--- @return success, redis_instance
+--- Method that gets a redis object after connection - it tries to authenticate based on an environment variable - REDIS_PASS or
+--- REDIS_PASSWORD
+function DefaultRedisConnectionProvider:getConnection(upstream)
```

Comment:
I think it would be cleaner if DefaultRedisConnectionProvider would be in its own file.